java/core/src/test/java/com/lancedb/lance/FragmentTest.java

```diff
@@ -21,9 +21,12 @@
 import java.nio.file.Path;
 import java.util.List;
 import java.util.Optional;
+
```

Comment:
nit on testing: let's add a few test cases for error handling? Just to make sure we aren't panicing or mishandling common errors.

Reply:
Added some scanner test cases, but yeah, still not enough test coverage. Will add more tests

Reply:
should we use `assertThrows` to capture more detailed exceptions?